packages/core/src/modules/ledger/ledgerUtil.ts

```diff
@@ -15,3 +16,32 @@ export function generateSchemaId(did: string, name: string, version: string) {
 export function generateCredentialDefinitionId(did: string, seqNo: number, tag: string) {
   return `${did}:3:CL:${seqNo}:${tag}`
 }
+
+export function generateCredDefFromTemplate(
+  credentialDefinitionId: string,
+  credentialDefinitionTemplate: CredentialDefinitionTemplate | Omit<CredentialDefinitionTemplate, 'signatureType'>
+): Indy.CredDef {
+  return {
+    id: credentialDefinitionId,
+    schemaId: credentialDefinitionTemplate.schema.id,
+    type: 'BaseRecord', // Irrelevant for qualifiedIdentifier construction
+    tag: credentialDefinitionTemplate.tag,
+    version: credentialDefinitionTemplate.schema.version,
+    ver: credentialDefinitionTemplate.schema.ver,
+    value: {
+      primary: { ...credentialDefinitionTemplate.schema },
+      revocation: credentialDefinitionTemplate.supportRevocation,
+    },
+  } as Indy.CredDef
+}
```

Comment:
Why is this needed? It creates an invalid object so would rather skip the step to go to cred def

Reply:
refactoring out. Indeed not needed. had this because some function was expecting credDef instead of template mainly because of the .id prop on it but we can just pass this as params instead of creating a dummie cred def from `Omit<CredentialDefinitionTemplate, "signatureType">`